tests/enterprise-contract/contract.go

```diff
@@ -0,0 +1,146 @@
+				PublicKey:           fmt.Sprintf("k8s://%s/%s", namespace, publicSecretName),
+				SSLCertDir:          "/var/run/secrets/kubernetes.io/serviceaccount",
+				Strict:              false,
+			}
+			secretName := fmt.Sprintf("golden-image-public-key%s", util.GenerateRandomString(10))
+			//The staging public key for verificaiton image
+			publicKey := []byte("-----BEGIN PUBLIC KEY-----\n" +
+				"MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEZP/0htjhVt2y0ohjgtIIgICOtQtA\n" +
+				"naYJRuLprwIv6FDhZ5yFjYUEtsmoNcW7rx2KM6FOXGsCX3BNc7qhHELT+g==\n" +
+				"-----END PUBLIC KEY-----")
+
+			GinkgoWriter.Println("Create golden image public signing key")
+			Expect(kubeController.CreateOrUpdateSigningSecret(publicKey, secretName, namespace)).To(Succeed())
+			generator.PublicKey = fmt.Sprintf("k8s://%s/%s", namespace, secretName)
+			pipelineRunTimeout = int(time.Duration(5) * time.Minute)
+
+		})
+
+		It("verifies the release policy: Task bundle is not acceptable", func() {
+			policy := ecp.EnterpriseContractPolicySpec{
+				Sources: policySource,
+				Configuration: &ecp.EnterpriseContractPolicyConfiguration{
+					Include: []string{"attestation_task_bundle.unacceptable_task_bundle"},
+				},
+			}
+			Expect(kubeController.CreateOrUpdatePolicyConfiguration(namespace, policy)).To(Succeed())
+
+			generator.Image = "quay.io/redhat-appstudio/ec-golden-image:08b938499ee47a25673f5e6a56f69ce57d87b45c"
+			pr, err := kubeController.RunPipeline(generator, pipelineRunTimeout)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(kubeController.WatchPipelineRun(pr.Name, pipelineRunTimeout)).To(Succeed())
+
+			pr, err = kubeController.Tektonctrl.GetPipelineRun(pr.Name, pr.Namespace)
+			Expect(err).NotTo(HaveOccurred())
+
+			tr, err := kubeController.GetTaskRunStatus(pr, "verify-enterprise-contract")
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(tr.Status.TaskRunResults).Should(ContainElements(
+				tekton.MatchTaskRunResultWithJSONPathValue("HACBS_TEST_OUTPUT", "{$.result}", `["FAILURE"]`),
+			))
+
+			//Get container step-report log details from pod
+			reportLog, err := kubeController.Commonctrl.GetContainerLogs(tr.Status.PodName, "step-report", namespace)
+			GinkgoWriter.Printf("*** Logs from pod '%s', container '%s':\n----- START -----%s----- END -----\n", tr.Status.PodName, "step-report", reportLog)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(reportLog).Should(ContainSubstring("msg: Pipeline task 'build-container' uses an unacceptable task bundle"))
+		})
+
+		It("verifies the release policy: Task bundle is out of date", func() {
+			policy := ecp.EnterpriseContractPolicySpec{
+				Sources: policySource,
+				Configuration: &ecp.EnterpriseContractPolicyConfiguration{
+					Include: []string{"attestation_task_bundle.out_of_date_task_bundle"},
+				},
+			}
+			Expect(kubeController.CreateOrUpdatePolicyConfiguration(namespace, policy)).To(Succeed())
+
+			generator.Image = "quay.io/redhat-appstudio/ec-golden-image:c2f9e02e272ff8da32133ebcff454560c9dd0e38"
```

Comment:
Eventually, the tasks used by this image will be considered unacceptable and this test will fail

Reply:
This is my only major concern with this PR.

Reply:
> This is my only major concern with this PR.

Maybe the para --effective-time is a solution?

Reply:
Yes! Excellent idea. We do need to expose that as a task parameter. Currently, it's just a cli parameter.

Reply:
@lcarva code updated based on comments, Could you PTAL? Thanks.